airbyte-integrations/connectors/source-intercom/source_intercom/source.py

```diff
@@ -227,13 +228,17 @@ def should_retry(self, response: requests.Response) -> bool:
         if self.check_exists_scroll(response):
             self._backoff_count += 1
             if not self.can_use_scroll():
-                self.logger.error("Can't create a new scroll request within an minute. " "Let's try to use a standard non-scroll endpoint.")
+                self.logger.error("Can't create a new scroll request within an minute or scroll param was expired. " 
+                                  "Let's try to use a standard non-scroll endpoint.")
                 return False
 
             return True
         return super().should_retry(response)
 
     def backoff_time(self, response: requests.Response) -> Optional[float]:
+        if response.status_code == 404:
+            self._backoff_count = 3
+            return 0.01
```

Comment:
maybe will we move "3" to a separate common variable?
Can we switch to the "standard" endpoint without delays (0.01c)?
Could you add a short comment why here we set this attempt to maximum value here?

Reply:
No, if backoff time will be 0. It means that we use `DefaultBackoffException`
```
custom_backoff_time = self.backoff_time(response)
            if custom_backoff_time:
                raise UserDefinedBackoffException(backoff=custom_backoff_time, request=request, response=response)
            else:
                raise DefaultBackoffException(request=request, response=response)
```

Changed scroll or standard mechanism.